### DISCUSSION TITLE
RBAC granular permissions to rollback application

### DISCUSSION THREAD

**Discussion by marcogenualdo** [argoproj/argo-cd#19846] [Q&A]
I have several applications on a Kubernetes cluster deployed using ArgoCD. One role needs to be able, among other things, to rollback a given application. I know that the line
```
p, role:default, applications, update, */*, allow
```
grants the rollback permission, but it also grants permission to change the application in any way (e.g. edit yaml values), which is undesirable. Is there a set of policies that allows me to grant a role only the rollback operation?

**Comment by tapas-clear** [comment]
I don't believe there is a way to have granular permissions specific for rollback (at least with v2.12) but I believe `sync` not `update` is what actually grants access to rollback. Therefore, granting sync and thus rollback permissions wouldn't allow access to editing the yaml values if that's the main concern.

```
p, role:default, applications, sync, */*, allow
```

Source code to why I think it's sync not update: https://github.com/argoproj/argo-cd/blob/master/server/application/application.go#L2042

**Comment by Sinhyeok** [comment]
For production deployments, we need to enforce strict controls. We want to prevent general sync operations for most users to avoid unintended deployments. However, for incident response, there's a critical need to allow only rollback to a previous stable version.

Currently, "rollback" functionality is tied to the general sync permission. We would like to request that rollback be separated into its own distinct RBAC action.

This would allow us to grant users or roles the permission to rollback an application without granting them the broader sync permission, which could be used to sync to HEAD or an arbitrary new version. Having a dedicated action: rollback (or similar) would significantly enhance our ability to implement fine-grained access control for production environments.
